src/ostree/ot-builtin-prune.c

```diff
@@ -239,6 +242,41 @@ ostree_builtin_prune (int argc, char **argv, GCancellable *cancellable, GError *
                                   cancellable, error))
         return FALSE;
 
+      /* Process --only-branch. Note this combines with --retain-branch-depth; one
+       * could do e.g.:
+       *  * --only-branch exampleos/x86_64/foo
+       *  * --retain-branch-depth exampleos/x86_64/foo=0
+       * to prune exampleos/x86_64/foo to just the latest commit.
+       */
+      if (opt_only_branches)
+        {
+          /* Turn --only-branch into a set */
+          g_autoptr(GHashTable) only_branches_set = g_hash_table_new (g_str_hash, g_str_equal);
+          for (char **iter = opt_only_branches; iter && *iter; iter++)
+            {
+              const char *ref = *iter;
+              g_autofree char *commit = NULL;
+              /* Ensure the specified branch exists */
+              if (!ostree_repo_resolve_rev (repo, ref, FALSE, &commit, error))
+                return FALSE;
+              g_hash_table_add (only_branches_set, *iter);
+            }
+
+          /* Iterate over all refs, add equivalent of --retain-branch-depth=$ref=-1
+           * if the ref isn't in --only-branch set.
+           */
+          g_hash_table_iter_init (&hash_iter, all_refs);
+          while (g_hash_table_iter_next (&hash_iter, &key, &value))
+            {
+              const char *ref = key;
+              if (!g_hash_table_contains (only_branches_set, ref))
+                g_hash_table_insert (retain_branch_depth, g_strdup (ref), GINT_TO_POINTER ((int)-1));
```

Comment:
Should we be erroring in case the ref is already in the hash table? E.g. someone  typing `--only-branch foo --retain-branch-depth=bar=5` might not understand what the command is doing.